### PR TITLE
Fix progress folder display for Exchange

### LINKED ISSUE #3013: [Bug]: Calendar Progress Bars show id, not name
### What happened?

Calendar backups and restores show the calendar ID, not the calendar display name.

### Corso Version?

Corso v6.1.0

### Where are you running Corso?

any

### Relevant log output

```shell
n/a
```

### PR DESCRIPTION
**Pull request by ashmrtn**
---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* closes #3013

#### Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/exchange/exchange_data_collection.go [hunk 1]
```diff
@@ -182,8 +182,7 @@ func (col *Collection) streamItems(ctx context.Context, errs *fault.Bus) {
 		colProgress, closer = observe.CollectionProgress(
 			ctx,
 			col.fullPath.Category().String(),
-			// TODO(keepers): conceal compliance in path, drop Hide()
-			clues.Hide(col.fullPath.Folder(false)))
+			col.LocationPath().Elements())
```
Comment (ashmrtn) [inline src/internal/connector/exchange/exchange_data_collection.go:185]: @ryanfkeepers is this the correct way to do logging for a `path.Builder`?

Comment (ryanfkeepers) [reply src/internal/connector/exchange/exchange_data_collection.go:185]: Yup.  Builders and elements already comply with clues.Concealer.

**Review by ryanfkeepers — APPROVED** [review]
generally looks good, can I get a manual output snapshot for verification, please?

**Comment by ashmrtn** [comment]
Test run for events only on a user
```text
go run ./corso.go backup create exchange --data events --user user@domain.com --retain-progress
Flag --user has been deprecated, use --mailbox instead
Connecting to repository:      …
Connecting to M365:         done 
Backing Up ∙ user@domain.co…
Discovering items to backup:   …
∙ events:      6s done 
Backing up data:     12s done 
events 143 - Corso_Restore_25-A…
events ∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙    1 - test2
events ∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙∙    2 - Calendar
events 140 - United States holi…
Done

  ID                                    Bytes Uploaded  Items Uploaded  Items Skipped  Errors
  03661b69-1597-4fcf-ac90-aaaaaaaaaaaa  439 kB          288             0              0     
 
Completed Backups:
  ID                                    Started At            Duration   Status     Resource Owner                 
  03661b69-1597-4fcf-ac90-aaaaaaaaaaaa  2023-04-28T22:33:30Z  20.56552s  Completed  user@domain.com
```

**Pull request merged**